File: Compiler/src/compiler/values/BooleanValue.java
package compiler.values;

import compiler.Value;
import util.CompilationResult;
import util.DataType;
import util.JenaUtil;

public class BooleanValue extends Value {
    public BooleanValue(Boolean value) {
        super(Boolean.toString(value));
    }

    @Override
    public DataType resultDataType() {
        return DataType.BOOLEAN;
    }

    @Override
    public CompilationResult compile() {
        return new CompilationResult(JenaUtil.genBooleanVal(value), "", "");
    }
}

File: Compiler/src/compiler/values/BooleanValue.kt
package compiler.values

import compiler.Value
import util.CompilationResult
import util.DataType
import util.JenaUtil

class BooleanValue(value: Boolean) : Value(value.toString()) {

    override var value = value.toString()
        set(value) {
            require(value == "true" || value == "false") { "Некорректное значение" }
            field = value
        }

    override fun resultDataType(): DataType = DataType.Boolean

    override fun compile(): CompilationResult =
        CompilationResult(JenaUtil.genBooleanVal(value), emptyList(), "")
}